src/providers/afdian.ts

```diff
@@ -0,0 +1,69 @@
+import { createHash } from 'node:crypto'
+import { $fetch } from 'ohmyfetch'
+import type { Provider, Sponsorship } from '../types'
+
+// afdian api docs https://afdian.net/p/9c65d9cc617011ed81c352540025c377
+
+export const AfdianProvider: Provider = {
+  name: 'afdian',
+  fetchSponsors(config) {
+    return fetchAfdianSponsors(config.afdian?.userId, config.afdian?.token)
+  },
+}
+
+export async function fetchAfdianSponsors(userId?: string, token?: string): Promise<Sponsorship[]> {
+  if (!userId || !token)
+    throw new Error('Afdian id and token are required')
+
+  const sponsors: any[] = []
+  const sponsorshipApi = 'https://afdian.net/api/open/query-sponsor'
```

Comment:
I have also implemented an afdian provider myself.

I used [afdian-api](https://github.com/backrunner/afdian-api). I think that might be more concise
> https://github.com/YunYouJun/sponsors/blob/master/kit/providers/afdian.ts

@antfu If you think that's better, I'd be happy to create a pr for that.

Reply:
Would the current code work for you? I would prefer not to add a new dependency if possible.

Reply:
That's OK. It works.